examples/data-objects/inventory-app/src/view/inventoryList.tsx

```diff
@@ -3,34 +3,138 @@
  * Licensed under the MIT License.
  */
 
-import { useTree } from "@fluid-experimental/tree-react-api";
+import {
+	objectIdNumber,
+	usePropTreeNode,
+	withTreeObservations,
+	withMemoizedTreeObservations,
+	type PropTreeNode,
+} from "@fluid-experimental/tree-react-api";
 import * as React from "react";
 
 import type { Inventory } from "../schema.js";
+import { Part } from "../schema.js";
 
 import { Counter } from "./counter.js";
 
-export const MainView: React.FC<{ root: Inventory }> = ({ root: inventory }) => {
-	useTree(inventory);
+/**
+ * Top level view is an easy place to enable StrictMode if desired.
+ *
+ * This component does not use any of the tree invalidation logic.
+ * This is safe because the node is passed in as a PropTreeNode:
+ * the type system will not allow this component to access any of the contents of the node that can change
+ * and thus no custom invalidation is needed.
+ */
+export const MainView: React.FC<{ root: PropTreeNode<Inventory> }> = ({ root }) => {
+	return (
+		// <React.StrictMode>
+		<InventoryView root={root} />
+		// </React.StrictMode>
+	);
+};
+
+/**
+ * View which consumes part of a tree, delegating some to sub-components.
+ *
+ * This demonstrates how arrays can be handled efficiently and easily.
+ */
+const InventoryView: React.FC<{ root: PropTreeNode<Inventory> }> =
+	withMemoizedTreeObservations(({ root }: { root: Inventory }) => {
+		const parts = root.parts.map((part) => (
```

Comment:
Nit: since your input includes `root.parts`, and this represents the resulting _views_ of those parts, might be good to call this `partViews` or something.